### PR TITLE
feat: clear out zombie images on comment html

### PR DESCRIPTION
**Pull request by sshanzel**
- Refactored the usage of `enum` to follow a PascalCase setup.
- New pubsub topics for:
  - Comment edited.
  - Comment deleted.
- New subscription for post commented and comment commented to update uploaded images.
- Created a generator function for creating workers related to clearing out content images.

### CODE REVIEW

File: src/workers/cdc.ts [hunk 1]
```diff
@@ -193,6 +195,14 @@ const onCommentChange = async (
         data.payload.after.id,
       );
     }
+  } else if (data.payload.op === 'u') {
+    await notifyCommentEdited(logger, data.payload.after);
```
Comment (idoshamun) [inline src/workers/cdc.ts:199]: Let's check here that it was the content that got edited

File: src/workers/cdc.ts [hunk 2]
```diff
@@ -193,6 +195,14 @@ const onCommentChange = async (
...
+  } else if (data.payload.op === 'd') {
+    await notifyCommentDeleted(logger, {
+      ...data.payload.before,
+      content: undefined,
```
Comment (idoshamun) [inline src/workers/cdc.ts:203]: Why do you want to change these props? I think we should keep it as is. We can't be 100% sure who's gonna subscribe to this topic, and this is not an expected behavior

File: __tests__/workers/commentMarkdownImages.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+beforeEach(async () => {
+  jest.resetAllMocks();
+  await con.getRepository(ContentImage).save([
+    { serviceId: '1', url: 'https://daily.dev/1.jpg' },
+    { serviceId: '2', url: 'https://daily.dev/2.jpg' },
+    { serviceId: '3', url: 'https://daily.dev/3.jpg' },
+  ]);
+});
+
+it('should do nothing for post without a content html', async () => {
```
Comment (idoshamun) [inline __tests__/workers/commentMarkdownImages.ts:34]:
```suggestion
it('should do nothing for comment without a content html', async () => {
```

**Review by idoshamun — CHANGES_REQUESTED** [review]
This looks great! Love the generators approach.
Please see my comments and you also have a pulumi error due to duplicate subscription names

File: src/workers/cdc.ts (hunk 2 above)
Comment (sshanzel) [reply src/workers/cdc.ts:203]: I agree, but we would need a reference on which comment was deleted. The `after` is `null` once deleted, so I had to take the `before` for the IDs and applied the empty content as part of being a `null` after.

File: src/workers/cdc.ts (hunk 1 above)
Comment (sshanzel) [reply src/workers/cdc.ts:199]: The only thing that can change from here is the `content` (and `contentHtml`) but I agree on making it explicit just to be sure 🚢

File: src/workers/index.ts [hunk 4]
```diff
@@ -40,6 +42,8 @@ export const workers: Worker[] = [
   deleteUserFromMailingList,
   commentUpvotedRep,
   commentUpvoteCanceledRep,
+  commentMarkdownImages,
```
Comment (idoshamun) [inline src/workers/index.ts:45]: I think you miss some workers here. I see more new subscriptions then workers here

Comment (sshanzel) [reply src/workers/index.ts:45]: Yes, I thought we can put 2 workers (edit and delete) in a single subscription. I am currently adding a dedicated one for deleting purposes.

File: src/workers/cdc.ts (hunk 2 above)
Comment (idoshamun) [reply src/workers/cdc.ts:205]: The topic itself means this comment is deleted. It's not trivial to mess with the properties of the data. Again, someone in the future may subscribe to event and will not figure out what happens here. I suggest we find another way to indicate deletion, e.g. providing a flag to the generator

Comment (sshanzel) [reply src/workers/cdc.ts:205]: The value of `data.payload.after` is `null`, the subscriber won't have any reference to which comment was deleted if we pass just the `after` object. We have two options:
- We will pass the `id` only in here to still have the reference on the subscription, or
- We will send a flag to the generator to clear **all** `ContentImage` with the `type` we sent with all the `referenceId` that is not present on the relevant table anymore. Might become a heavy process in the future when as we will fetch all `Comment` type (for example)  then we will query all reference IDs that do not exist on the `Comments` table anymore.

Comment (idoshamun) [reply src/workers/cdc.ts:205]: I think you misunderstood me. My suggestion is to send `before` as you did already but without manipulating `content` and `contentHtml`. Just pass the object as is

Comment (sshanzel) [reply src/workers/cdc.ts:205]: Oh, right. Yep. That makes sense. Let me do that 🚀

**Comment by sshanzel** [comment]
I have now added an optional parameter to allow the generator to make some decisions based on some values and to keep the openness of the function to changes 🚢

**Review by idoshamun — APPROVED** [review]

**Review by rebelchris — APPROVED** [review]
Looks amazing!

**Pull request merged**